java/code/src/com/redhat/rhn/domain/access/AccessGroupFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2025 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.redhat.rhn.domain.access;
+
+import com.redhat.rhn.common.hibernate.HibernateFactory;
+import com.redhat.rhn.domain.org.Org;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Factory class for RBAC's {@link AccessGroup} entities
+ */
+public class AccessGroupFactory extends HibernateFactory {
+
+    private static final AccessGroupFactory INSTANCE = new AccessGroupFactory();
+    private static final Logger LOG = LogManager.getLogger(AccessGroupFactory.class);
+
+    private AccessGroupFactory() {
+        super();
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return LOG;
+    }
+
+    /**
+     * Persists an {@code AccessGroup} entity to DB.
+     * @param accessGroupIn the entity to save
+     * @return the saved entity
+     */
+    public static AccessGroup save(AccessGroup accessGroupIn) {
+        INSTANCE.saveObject(accessGroupIn);
+        return accessGroupIn;
+    }
+
+    /**
+     * Deletes an {@code AccessGroup} entity from DB.
+     * @param accessGroupIn the entity to remove
+     */
+    public static void remove(AccessGroup accessGroupIn) {
+        INSTANCE.removeObject(accessGroupIn);
+    }
+
+    /**
+     * Lists all access groups defined in MLM.
+     * @return the list of access groups
+     */
+    public static List<AccessGroup> listAll() {
+        return getSession()
+                .createQuery("SELECT a FROM AccessGroup a", AccessGroup.class)
+                .getResultList();
+    }
+
+    /**
+     * Lists all access groups that belong to an org.
+     * @param org the org
+     * @return the list of access groups
+     */
+    public static List<AccessGroup> list(Org org) {
+        return getSession()
+                .createQuery("SELECT a FROM AccessGroup a WHERE a.org = :org OR a.org IS NULL", AccessGroup.class)
```

Comment:
This query is not really returning all groups that belong to an Org. In fact it returns all groups from the org or the default ones (not linked to any org).

Reply:
Yes, it returns all the groups available to an org, including the defaults, which are collectively owned by every org.